lib/iris/tests/unit/fileformats/netcdf/test_Saver__ugrid.py

```diff
@@ -0,0 +1,1171 @@
+            list(dims.keys()),
+            ["Mesh1d_node", "Mesh1d_edge", "Mesh1d_edge_N_nodes"],
+        )
+        self.assertEqual(vars_meshdim(vars, "node"), "Mesh1d_node")
+        self.assertEqual(vars_meshdim(vars, "edge"), "Mesh1d_edge")
+
+        # check suitable mesh properties
+        self.assertEqual(mesh_name, "Mesh1d")
+        mesh_props = vars[mesh_name]
+        self.assertEqual(mesh_props["topology_dimension"], 1)
+        self.assertIn("edge_node_connectivity", mesh_props)
+        self.assertNotIn("face_node_connectivity", mesh_props)
+
+    def test_location_coord_units(self):
+        # Check that units on mesh locations are handled correctly.
+        # NOTE: at present, the Mesh class cannot handle coordinates that are
+        # not recognised by 'guess_coord_axis' == suitable standard names
+        mesh = make_mesh(
+            nodecoord_xyargs=(
+                {
+                    "standard_name": "projection_x_coordinate",
+                    "var_name": "node_x",
+                    "units": "degrees",  # should NOT convert to 'degrees_east'
+                    "axis": "x",  # N.B. this is quietly dropped !!
+                },
+                {
+                    "standard_name": "projection_y_coordinate",
+                    "var_name": "node_y",
+                    "units": "ms-1",
+                    "axis": "y",  # N.B. this is quietly dropped !!
+                },
+            ),
+            facecoord_xyargs=(
+                {
+                    "standard_name": "longitude",
+                    "var_name": "face_x",
+                    "units": "",  # SHOULD result in no units property
+                },
+                {
+                    "standard_name": "latitude",
+                    "var_name": "face_y",  # SHOULD convert to 'degrees_north'
+                    "units": "degrees",
+                },
+            ),
+        )
+
+        # Save and snapshot the result
+        tempfile_path = self.check_save_mesh(mesh)
+        dims, vars = scan_dataset(tempfile_path)
+
+        # there is a single mesh-var
+        (mesh_name,) = vars_meshnames(vars)
+
+        # find the node- and face-coordinate variables
+        node_x = vars["node_x"]
+        node_y = vars["node_y"]
+        face_x = vars["face_x"]
+        face_y = vars["face_y"]
+
+        # Check that units are as expected.
+        # 1. 'long/lat' degree units are converted to east/north
+        # 2. non- (plain) lonlat are NOT converted
+        # 3. other names remain as whatever was given
+        # 4. no units on input --> none on output
+        self.assertEqual(node_x["units"], "degrees")
+        self.assertEqual(node_y["units"], "ms-1")
+        self.assertNotIn("units", face_x)
+        self.assertEqual(face_y["units"], "degrees_north")
+
+        # N.B. we also do not automatically add 'axis' properties, nor recognise
+        # them on input :  Because we *only* do that for dim-coords.
```

Comment:
@pp-mo Just to clarify... If I recall (could be mistaken) but we don't ever store the `axis`, we only infer it, right?

So this comment about dim coords is a little misleading... or am I confused / missing something?